### ISSUE TITLE
Implement a SagaManagerListener

### ISSUE THREAD

**Issue by cer** [eventuate-tram/eventuate-tram-sagas#98]
SagaManager invoke the listener as the state of a saga changes: started, step executed, ..., completed.

One possible implementation could publish domain events.

_Originally posted by @cer in https://github.com/eventuate-tram/eventuate-tram-sagas/issues/95#issuecomment-1473924479_

**Comment by kumar-csice** [comment]
@cer 
Is this SagaManager provides features like orchestration GUI tool/service for the monitoring this eventuate orchestration flow ?

**Comment by cer** [comment]
No.  It's a [class](https://github.com/eventuate-tram/eventuate-tram-sagas/blob/master/eventuate-tram-sagas-orchestration/src/main/java/io/eventuate/tram/sagas/orchestration/SagaManagerImpl.java) that implements much of the saga orchestration logic.
